events/events.go

```diff
@@ -0,0 +1,261 @@
+		client = httpClientEnabledSecurityCheck
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	return resp, err
+}
+
+func executeCurl(apiUrl string, accessToken string, path string) ([]map[string]interface{}, error) {
+	headers := make(map[string]string)
+	headers["Content-Type"] = "application/json"
+	headers["Accept"] = "application/json"
+	headers["Authorization"] = "bearer " + accessToken
+	var decodedBodyArray []map[string]interface{}
+	hasNextUrl := true
+	url := apiUrl + path
+	for hasNextUrl {
+		curlResponse, err := callHttpMethod("GET", url, headers, nil, true)
+		defer curlResponse.Body.Close()
+		var decodedBody map[string]interface{}
+		if err != nil {
+			fmt.Printf("Error while CURL call")
+			return decodedBodyArray, err
+		} else {
+			bodyBytes, err2 := ioutil.ReadAll(curlResponse.Body)
+			if err2 != nil {
+				fmt.Printf("Error while decoding curl response")
+				return decodedBodyArray, err2
+			}
+
+			err = json.Unmarshal(bodyBytes, &decodedBody)
+			if err != nil {
+				return decodedBodyArray, err
+			}
+			nextUrl := decodedBody["next_url"]
+			if nextUrl != nil {
+				nextUrl := decodedBody["next_url"].(string)
+				url = apiUrl + nextUrl
+			} else {
+				hasNextUrl = false
+			}
+			decodedBodyArray = append(decodedBodyArray, decodedBody)
+
+		}
+	}
+	return decodedBodyArray, nil
+}
+
+func getAccessToken(loginUrl string, refreshToken string, grantType string) (string, error) {
+	headers := make(map[string]string)
+	headers["Content-Type"] = "application/x-www-form-urlencoded"
+	headers["Accept"] = "application/json"
+	headers["Authorization"] = "Basic Y2Y6"
+
+	data := "grant_type=" + grantType + "&client_id=cf&client_secret=&refresh_token=" + refreshToken
+
+	tokenResponse, err := callHttpMethod("POST", loginUrl+"/oauth/token", headers, strings.NewReader(data), true)
+	defer tokenResponse.Body.Close()
+	if err != nil {
+		fmt.Printf("Error while getting access-token (CF-Login-CURL call)")
+		return "", err
+	} else {
+		bodyBytes, err2 := ioutil.ReadAll(tokenResponse.Body)
+		if err2 != nil {
+			fmt.Printf("Error while decoding curl response")
+			return "", err2
+		}
+		var s map[string]string
+		err = json.Unmarshal(bodyBytes, &s)
+		accessToken := s["access_token"]
+		return accessToken, nil
+	}
+}
+
+func (c *EventCommand) ListEvents(cliConnection plugin.CliConnection, noInstanceNames bool, action string) {
+	initialize()
+	fmt.Println("Getting the list of instance events in the org", AddColor(helper.GetOrgName(helper.ReadConfigJsonFile()), constants.Cyan), "/ space", AddColor(helper.GetSpaceName(helper.ReadConfigJsonFile()), constants.Cyan), "...")
+	var cmd string
+	var guid string
+	var instanceName string
+	var eventType string
+	var user string
+	var createTime string
+	var actionType string
+	var userSpaceGuid = helper.GetSpaceGUID(helper.ReadConfigJsonFile())
+
+	if helper.GetAccessToken(helper.ReadConfigJsonFile()) == "" {
+		errors.NoAccessTokenError("Access Token")
+	}
```

Comment:
This is just for a basic check for verification of access_token in config.json file.

Reply:
Since you are using `GetRefreshToken` you can check for that instead of `GetAccessToken`, or since your access token call may already take care of that, you can remove it.